### PR TITLE
Newlines are now ignored while splitting by sentences in summarizing. Fix #1575.

### LINKED ISSUE #1575: Add preprocessing in summarize function to remove newlines in the middle of sentences
#### Description
The `summarize` function depends on the text having at least 10 sentences as measured by `clean_text_by_sentences`.  If the text is shorter than that then the summarization fails in an undocumented way. Moreover  `clean_text_by_sentences` cannot handle properly a text with new lines
at the middle of a sentence. I suggest a preprocessing step to purge those.

I'm currently using this to workaround this bug
```
import re
text = re.sub(r'\n|\r|\t', ' ', text)
text = re.sub(r'\s+', ' ', text)
```

#### Versions
- Python 3.6.2 (default, Jul 17 2017, 16:44:45)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
- NumPy 1.13.0
- SciPy 0.19.1
- gensim 2.3.0
- FAST_VERSION 0

### PR DESCRIPTION
**Pull request by fbarrios**
This is an improvement over #1577 for fixing #1575, as this doesn't need to process all text at once for removing newlines.

### CODE REVIEW

File: gensim/test/test_data/mihalcea_tarau.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-AP880911-0016
-AP-NR-09-11-88 0423EDT r i
-BC-HurricaneGilbert 09-11 0339
-BC-Hurricane Gilbert,0348
-Hurricane Gilbert Heads Toward Dominican Coast
-By RUDDY GONZALEZ
-Associated Press Writer
-SANTO DOMINGO, Dominican Republic (AP)
+AP880911-0016.
+AP-NR-09-11-88 0423EDT r i.
```
Comment (menshikh-iv) [inline gensim/test/test_data/mihalcea_tarau.txt]: Without dots, this doesn't work? You think that this is a good solution?

Comment (fbarrios) [reply gensim/test/test_data/mihalcea_tarau.txt]: Not really, but this was reported in #1575 and I created this PR as an improvement in performance over #1577.

I do believe that we should at least write about this behavior in the docstring.

**Comment by menshikh-iv** [comment]
Ping @fbarrios, how is going?

**Comment by menshikh-iv** [comment]
ping @fbarrios, what's status here?

**Comment by fbarrios** [comment]
Sorry @menshikh-iv, I've been busy this week. I'll look into this in the next few days.

**Comment by menshikh-iv** [comment]
Ping @fbarrios

**Comment by fbarrios** [comment]
@menshikh-iv I think the correct behaviour is the current one, so this PR should be closed, so please do if you agree.

I created #1702 for updating the docstring with the current behaviour.

**Pull request closed without merging**